Essentials/src/com/earth2me/essentials/commands/Commandmute.java

```diff
@@ -40,20 +41,29 @@ public void run(final Server server, final CommandSource sender, final String co
                 throw new Exception(tl("muteExempt"));
             }
         }
+
+        long muteTimestamp = 0;
+        String time;
+
+        if (args.length > 1) {
+            time = args[1];
+            muteTimestamp = DateUtil.parseDateDiff(time, true);
+            final long maxMuteLength = ess.getSettings().getMaxMute() * 1000;
+            if (maxMuteLength > 0 && ((muteTimestamp - GregorianCalendar.getInstance().getTimeInMillis()) > maxMuteLength) && sender.isPlayer() && !(ess.getUser(sender.getPlayer()).isAuthorized("essentials.mute.unlimited"))) {
```

Comment:
Why are we messing around with the Gregorian Calendar code? getTimeInMillis reports time as UTC millis from epoch - this doesn't account for time zone (nor should it).

If you are going to go with the calendar approach, go all the way and get rid of epoch time (2036!!!!)

Reply:
Side note I hate this pattern of sender.isPlayer _then_ doing a permission check - iirc console is op and will pass this permission if configured to pass the permission as op (which a user can override).

Nothing actionable, just a nit on all of essentials.

Reply:
@Ichbinjoe `User#isAuthorized` requires a user instance for a player, hence the need for the isPlayer check. It's not tidy, but it's a necessary part of Essentials' current permissions infrastructure which gives us permission logging and enables us to use `player-commands`.

Reply:
I mean I realize that - I'm just saying it makes me sad as an aside.

Reply:
On a similar vein to my last comment -- the check is borrowed from Commandtempban.java, which is proven to work. I was curious as to why it used GregorianCalendar as well, but didn't look into it too deeply. Again, if it's an issue this should probably be looked into more.